### PR TITLE
Fixes nginx errors and re-adds CORS support

### PR DESCRIPTION
**Pull request by pchojnacki**
This change fixes potential errors caused by malformed upstreams.conf

Also Most of the CORS functionality was implemented previously but the filter wasn't enabled so it wasn't working.

Here I reintroduce CORS support with additional headers and setting making api-gateway the authoritative entity WRT to CORS headers. 

As API-Gateway *will* override all existing cors headers set by services. In an effort to standardize the CORS policies across services.

/cc: @kvas-damian @Wikia/services-team @jimmordino @ArturKlajnerok

### CODE REVIEW

**Comment by pchojnacki** [comment]
this change is already deployed on production to provide functionality required by @kvas-damian

**Comment by kvas-damian** [comment]
Some background for other code reviewers. We introduced ```withCredentials``` flag for Cross domain XHRs to be able to store cookies returned by Helios.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Access-Control-Allow-Credentials

Works on production!

File: src/cors.lua [hunk 1]
```diff
@@ -23,8 +32,18 @@ function cors.origin_matches_whitelist(origin_value)
   return false
 end
 
+function cors.get_origin_that_matches_whitelist(ngx)
+  local req_headers = ngx.req.get_headers(20)
```
Comment (jimmordino) [inline src/cors.lua]: There was talk of increasing this limit or removing it altogether on the services channel.  Apparently, there are already > 20 headers in the request.

File: templates/conf/upstreams.conf [hunk 2]
```diff
@@ -5,10 +5,13 @@
 {{ if $service.Name }}{{if $tag}}
 {{ with $serviceQuery := printf "%s.%s" $tag $service.Name }}
 {{ if service $serviceQuery }}
+# query: {{$serviceQuery}}
+{{ with $queryResult := service $serviceQuery }}{{ if $queryResult }}
 upstream {{template "escape" $serviceQuery}} {
     {{range service $serviceQuery}}
         server {{.Address}}:{{.Port}};{{end}}
 }
+{{end}}{{end}}
```
Comment (drsnyder) [inline templates/conf/upstreams.conf]: :+1:

**Comment by drsnyder** [comment]
LGTM other than the comment from @jimmordino about the headers. It's probably fine to grab all the headers.

File: src/cors.lua (hunk 1 above)
Comment (pchojnacki) [inline src/cors.lua]: Thanks for pointing it out. I'll increase it to 100

**Pull request merged**